avro/src/main/scala/AvroRandomGenerator.scala

```diff
@@ -0,0 +1,41 @@
+package com.memsql.spark.examples.avro
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericData
+
+import scala.collection.JavaConversions._
+import scala.util.Random
+
+class AvroRandomGenerator(inSchema: Schema) {
+  val topSchema: Schema = inSchema
+  val random = new Random
+  val MAX_RECURSION_LEVEL:Int = 5
+
+  def next(schema: Schema = this.topSchema, level: Int = 0): Any = {
+    if (level > MAX_RECURSION_LEVEL) {
+      return null
+    }
+
+    schema.getType match {
+      case Schema.Type.BOOLEAN => random.nextBoolean
+      case Schema.Type.DOUBLE => random.nextDouble
+      case Schema.Type.FLOAT => random.nextFloat
+      case Schema.Type.INT => random.nextInt
+      case Schema.Type.LONG => random.nextLong
+      case Schema.Type.NULL => null
+      case Schema.Type.RECORD => {
+        val datum = new GenericData.Record(schema)
```

Comment:
No - this creates a record based on the schema passed in and creates random values for every field on that record.